### PR TITLE
test: add apply-mapping test condition for custom fields in data file

### PR DESCRIPTION
**Pull request by johnbieren**

### CODE REVIEW

**Review by happybhati — APPROVED** [review]

File: tasks/apply-mapping/tests/test-apply-mapping.yaml [hunk 1]
```diff
@@ -125,6 +125,10 @@ spec:
               test $(cat $(workspaces.config.path)/test_snapshot_spec.json \
                 | jq -r '.components[] | select(.name=="comp4") | .repository') == repo4
 
+              echo Test that customfield of component comp4 was added to its snapshot entry
```
Comment (mmalina) [inline tasks/apply-mapping/tests/test-apply-mapping.yaml:128]: Just curious, how did you think of this? What made you add this now? And what's the use case for having a custom field anyway?

Comment (johnbieren) [reply tasks/apply-mapping/tests/test-apply-mapping.yaml:128]: I wanted to make sure `pushSourceContainer: true` on a component was propagated to the resulting snapshot_spec file

Comment (mmalina) [reply tasks/apply-mapping/tests/test-apply-mapping.yaml:128]: ok, thanks

**Pull request merged**